src/main/java/com/example/glovo/service/OrderService.java

```diff
@@ -0,0 +1,75 @@
+package com.example.glovo.service;
+
+import com.example.glovo.model.Order;
+import com.example.glovo.model.Product;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@AllArgsConstructor
+@Service
+public class OrderService {
+    private static final Map<Long, Order> orders = new HashMap<>();
+    private static long orderIdCounter = 0;
+    private final ProductService productService;
+
+    public Order get(long id) {
+        return Optional.ofNullable(orders.get(id))
+                .orElseThrow(() -> new NoSuchElementException("Order with ID " + id + " does not exist"));
```

Comment:
optional is not suitable here, in this case, return Optional<Order> or remove optional